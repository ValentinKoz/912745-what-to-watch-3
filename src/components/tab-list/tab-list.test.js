import React from "react";
import renderer from "react-test-renderer";
import TabList from "./tab-list.jsx";

const film = {
  genre: `Жанр`,
  title: `Title`,
  releaseDate: 0,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  bgPoster: `img/moonrise-kingdom.jpg`,
  ratingScore: `8,9`,
  ratingLevel: `level`,
  ratingCount: 240,
  text: `текст...`,
  director: `Director`,
  runTime: `0h 00m`,
  starring: `starring`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  reviews: [{
    text: `reviews`,
    author: `Kate Muir`,
    time: `December 24, 2016`,
    rating: `8,9`
  }]
};

it(`Render list tabs correctly`, () => {
  const tree = renderer
    .create(
        <TabList
          film={film}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
