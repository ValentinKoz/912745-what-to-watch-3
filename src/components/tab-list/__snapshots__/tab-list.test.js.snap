// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render list tabs correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8,9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        level
      </span>
      <span
        className="movie-rating__count"
      >
        240 ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      текст...
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: starring
      </strong>
    </p>
  </div>
</div>
`;
